api/data/tacoma_db.go

```diff
@@ -7,16 +7,44 @@ import (
 	"github.com/jackc/pgx/v4"
 )
 
-type tacomaOfficer struct {
+// TacomaOfficer is the object model for Tacoma PD officers
+type TacomaOfficer struct {
 	FirstName  string       `json:"first_name,omitempty"`
 	LastName   string       `json:"last_name,omitempty"`
 	Title      string       `json:"title,omitempty"`
 	Department string       `json:"department,omitempty"`
 	Salary     nulls.String `json:"salary,omitempty"`
 }
 
-func (db *dbClient) tacomaSearchOfficerByName(firstName, lastName string) ([]*tacomaOfficer, error) {
-	rows, err := db.pool.Query(context.Background(), `
+// TacomaOfficerMetadata retreives metadata describing the TacomaOfficer struct
+func (c *Client) TacomaOfficerMetadata() []map[string]string {
+	return []map[string]string{
+		{
+			"FieldName": "first_name",
+			"Label":     "First Name",
+		},
+		{
+			"FieldName": "last_name",
+			"Label":     "Last Name",
+		},
+		{
+			"FieldName": "title",
+			"Label":     "Title",
+		},
+		{
+			"FieldName": "department",
+			"Label":     "Department",
+		},
+		{
+			"FieldName": "salary",
+			"Label":     "Salary",
+		},
+	}
+}
+
+// TacomaSearchOfficerByName invokes tacoma_search_officer_by_name_p
+func (c *Client) TacomaSearchOfficerByName(firstName, lastName string) ([]*TacomaOfficer, error) {
```

Comment:
This might be a Go thing - do you not have to define the type for all function arguments if they're all the same?

Reply:
Yeah you can roll them all up like that.